ui/src/app/pages/data/concept/concept-homepage.tsx

```diff
@@ -388,24 +366,28 @@ export const ConceptHomepage = withCurrentWorkspace()(
       this.setState({
         currentInputString: '',
         currentSearchString: '',
+        selectedSurvey: '',
         showSearchError: false,
         searching: false // reset the search result table to show browse/domain cards instead
       });
     }
 
-    browseDomain(domain?: DomainInfo) {
+    browseDomain(domain: DomainInfo) {
       const {conceptDomainCounts} = this.state;
-      const selectedDomain = !domain ? {domain: Domain.SURVEY, name: 'Surveys', conceptCount: 0}
-        : conceptDomainCounts.find(domainCount => domainCount.domain === domain.domain);
-      this.setState({currentInputString: '', currentSearchString: '', selectedDomain: selectedDomain}, () => this.searchConcepts());
+      const selectedDomain = conceptDomainCounts.find(domainCount => domainCount.domain === domain.domain);
+      this.setState(
+        {currentInputString: '', currentSearchString: '', selectedDomain: selectedDomain, selectedSurvey: ''},
+        () => this.searchConcepts()
+      );
     }
 
     browseSurvey(surveyName) {
-      if (environment.enableNewConceptTabs) {
-        this.browseDomain();
-      } else {
-        this.setState({browsingSurvey: true, selectedSurvey: surveyName});
-      }
+      this.setState({
+        currentInputString: '',
+        currentSearchString: '',
+        selectedDomain: {domain: Domain.SURVEY, name: 'Surveys', conceptCount: 0},
```

Comment:
Should conceptCount be dynamic instead of 0?

Reply:
It's basically a placeholder at this point. It will get replaced with the real count when `searchConcepts()` runs below.

Reply:
Oh.. gotcha. I think this is so, because surveys doesn't exist in the the domain_info table.